### PR TITLE
parallelize set names from tags

### PR DESCRIPTION
**Pull request by breunigs**
It uses the previously introduced `auto_modulo` to parallelize all update queries. It saves ~45% on the wall clock time in `set_names_from_tags.sql` for Hamburg metropolitan area, across 4 runs on this dataset. I did not confirm this for a larger area, as the trend seems pretty clear judging from the small sample here and similar optimization I ran on the larger dataset (i.e. https://github.com/OSMNames/OSMNames/commit/29bbc538ce903f8998e953c4ce675b89e5fe334c ).

### CODE REVIEW

**Comment by breunigs** [comment]
For reasons unclear to me, it *also* speeds up `set_street_ids_by_street_name.sql`. This is not a one-off, the results are consistent across multiple runs and different computers. I've put my timings here, the big table is best viewed in "raw" to be able to see the whole rows at once: https://gist.github.com/breunigs/d169e1450a9d645afcf5a02822a78650 . 

The DB tables in question order rows differently between runs, presumably because imposm doesn't guarantee order. This makes comparison annoying since `id` and `parent_id` will be different, so I exported just the `osm_id`, `name` and `alternative_name` columns and compared those. They are identical across runs. 

I swapped the order for master/PR for one of the runs, to ensure there is no "cache warming" or "CPU overheating" effect going on. It makes no difference.

I also put `/var/lib/postgresql/data` on a tmpfs to exclude disk layout as a contributing factor, but the unexpected speedup on `set_street_ids_by_street_name.sql` remains. I checked a couple of times I didn't accidentally include https://github.com/OSMNames/OSMNames/pull/225 in this PR which would explain the speedup.

The geonames export also looks okay. The assignment of housenumbers to streets is unstable (they get put on different "segments" of the same street), so I just did an eyeball comparison -- no obvious breakage either.

The query plans for the three `UPDATE` queries in `set_street_ids_by_street_name.sql` are largely the same (https://gist.github.com/breunigs/08d9fe757fcded5e8949b44476bb6f67) with one notable exception: the first query doesn't use an extra "Memoize" step, but directly runs the index scan. The timing differences are not visible when I run these manually. Vacuum was enabled for all runs for consistency, so the table statistics should be accurate for all runs. 

So to sum it up, I have no idea what causes the speedup. But it's a speedup, so let's not look a gift horse into the mouth?

**Comment by sfkeller** [comment]
> So to sum it up, I have no idea what causes the speedup. But it's a speedup, so let's not look a gift horse into the mouth?

1st rule of programming "if it works, don't touch it" :-)

Just kidding. You're doing impressive work here!

**Review by philippks — APPROVED** [review]
> So to sum it up, I have no idea what causes the speedup. But it's a speedup, so let's not look a gift horse into the mouth?

sounds reasonable 😄 thanks for the investigations!

> The assignment of housenumbers to streets is unstable (they get put on different "segments" of the same street)

I noticed this as well. Really unpleasant...
